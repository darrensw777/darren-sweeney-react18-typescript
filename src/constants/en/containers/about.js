export const ABOUT_COPY = {
    pageTitle: {
        title: "About me",
        subTitle:
            "TLDR; Living in Lincolnshire with my wife and our two boys, most of my work is taken up with freelance / contract projects. The main areas of work I concentrate on are web development contracts.",
    },
    aboutMe: {
        greetings: {
            morning: "Good morning.",
            evening: "Good evening.",
            afternoon: "Good afternoon.",
        },
        general: [
            "I currently work in the UK as a front end web developer.",
            "In past lives I've been a singer, a welder, worked in retail, and I'm Series 7 Licenced (If you know, you know...)",
            "I strive to create web sites which are as per design, and efficient.",
        ],
        interests: [
            {
                question: "Music",
                answers: ["Journey", "Steve Perry", "80s AOR", "Simon & Garfunkel", "Keane"],
            },
            {
                question: "TV Shows",
                answers: [
                    "Suits",
                    "Breaking Bad",
                    "The Witcher",
                    "Peaky Blinders",
                    "Mad Men",
                    "The Boys",
                    "Killing Eve",
                ],
            },
            {
                question: "Movies",
                answers: ["Goodfellas", "Wall St", "Gangs of New York", "The Wolf of Wall St"],
            },
            {
                question: "Sports",
                answers: ["Golf"],
            },
            {
                question: "Favourite football team",
                answers: ["Golf (see Sports)"],
            },
            {
                question: "Hobbies",
                answers: ["Golf", "Music (singing)", "Technology"],
            },
        ],
    },
};
