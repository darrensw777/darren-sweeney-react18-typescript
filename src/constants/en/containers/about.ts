export const ABOUT_COPY = {
    pageTitle: {
        title: "About me",
        subTitle:
            "TLDR; Living in Lincolnshire with my family, most of my work is taken up with JavaScript software development.",
    },
    aboutMe: {
        greetings: {
            morning: "Good morning.",
            evening: "Good evening.",
            afternoon: "Good afternoon.",
        },
        general: [
            "I currently work in the UK as a software developer.",
            "In past lives I've been a singer, a welder, worked in retail, and I'm Series 7 Licenced (If you know, you know...)",
            "I strive to create web sites which are as per design, and efficient.",
        ],
        interests: [
            {
                question: "Music",
                answers: ["Journey", "Steve Perry", "80s AOR", "Simon & Garfunkel", "Keane"],
            },
            {
                question: "TV Shows",
                answers: [
                    "Suits",
                    "Breaking Bad",
                    "The Witcher",
                    "Peaky Blinders",
                    "Mad Men",
                    "The Boys",
                    "Killing Eve",
                ],
            },
            {
                question: "Movies",
                answers: ["Goodfellas", "Wall St", "Gangs of New York", "The Wolf of Wall St"],
            },
            {
                question: "Sports",
                answers: ["Golf"],
            },
            {
                question: "Favourite football team",
                answers: ["Golf (see Sports)"],
            },
            {
                question: "Hobbies",
                answers: ["Golf", "Music (singing)", "Technology"],
            },
        ],
    },
};
