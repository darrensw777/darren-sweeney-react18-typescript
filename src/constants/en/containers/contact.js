export const CONTACT_COPY = {
    pageTitle: {
        title: "Contact me",
        subTitle: "Please use the form below to contact me and I'll reply as soon as I possibly can. Thanks!",
    },
    fieldLabels: {
        name: "Name:",
        email: "Email address:",
        message: "Message:",
    },
    misc: {
        allFieldsRequired: "All fields are required",
    },
    formSubmit: {
        formSuccess: "Success! Thanks, I'll reply as soon as I can.",
        formError: "Error: Please Try Again!",
    },
    validationErrorMessages: {
        nameRequired: "Please add your name",
        nameTooShort: "Please add more than one character",
        nameTooLong: "Maximum 50 characters",
        emailRequired: "Please add your email address",
        emailInvalid: "That email address is invalid",
        messageRequired: "Please add a message",
    },
};
