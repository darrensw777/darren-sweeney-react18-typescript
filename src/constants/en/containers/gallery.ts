export const GALLERY_COPY = {
    pageTitle: {
        title: 'Gallery',
        subTitle: 'Pictures of me and my favourite people!',
    },
};
