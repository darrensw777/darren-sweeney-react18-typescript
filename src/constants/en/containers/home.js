export const HOME_COPY = {
    pageTitle: {
        title: 'Home',
        subTitle:
            'Father, husband, web developer, singer. Have a look around and send me a message if you want to contact me.',
    },
};
