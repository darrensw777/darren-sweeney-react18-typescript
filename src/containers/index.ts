/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Home } from './Home';
export { default as About } from './About';
export { default as Work } from './Work';
export { default as Music } from './Music';
export { default as Gallery } from './Gallery';
export { default as Contact } from './Contact';
