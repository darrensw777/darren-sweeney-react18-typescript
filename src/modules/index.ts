/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
