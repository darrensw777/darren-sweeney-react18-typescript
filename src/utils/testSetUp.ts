import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
