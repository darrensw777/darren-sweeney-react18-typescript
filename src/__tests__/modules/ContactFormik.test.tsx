import { ContactFormik } from 'modules';
import { CONTACT_COPY } from 'constants/en/containers/contact';
import testSetUp from 'utils/testSetUp';

describe('ContactFormik container', () => {
    let getByText: any, asFragment: any, container: any, getByRole: any, getAllByRole: any;
    beforeEach(() => {
        ({ asFragment, getByText, container, getByRole, getAllByRole } = testSetUp({
            Component: <ContactFormik CONTACT_COPY={CONTACT_COPY} />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the block', () => {
        const blocks = container.getElementsByClassName('form-wrapper');
        expect(blocks.length).toBe(1);
    });
});
