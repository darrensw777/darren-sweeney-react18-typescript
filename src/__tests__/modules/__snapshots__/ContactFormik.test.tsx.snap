// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactFormik container matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="moduleStyles__ContactFormikWrapper-sc-187oehp-2 gqMlFz"
  >
    <div
      class="form-wrapper"
    >
      <form
        class="form"
        data-netlify="true"
        data-netlify-honeypot="bot-field"
        name="contactForm"
      >
        <input
          hidden=""
          name="bot-field"
        />
        <p
          class="helper"
        >
          All fields are required
        </p>
        <div
          class="formElementsStyles__InputFieldWrapper-sc-5xvoh1-2 eIaxFt"
        >
          <div
            class="field"
          >
            <label
              class="label"
              for="name"
            >
              Name:
            </label>
            <input
              class="input-text"
              id="name"
              name="name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="formElementsStyles__InputFieldWrapper-sc-5xvoh1-2 eIaxFt"
        >
          <div
            class="field"
          >
            <label
              class="label"
              for="name"
            >
              Email address:
            </label>
            <input
              class="input-text"
              id="email"
              name="email"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="formElementsStyles__InputFieldWrapper-sc-5xvoh1-2 eIaxFt"
        >
          <div
            class="field"
          >
            <label
              class="label"
              for="name"
            >
              Message:
            </label>
            <textarea
              class="input-textarea"
              id="message"
              name="message"
              rows="6"
              type="text"
            />
          </div>
        </div>
        <div
          class="buttons"
        >
          <div
            class="formElementsStyles__ButtonWrapper-sc-5xvoh1-1 karwgk"
          >
            <button
              aria-label="Send message"
              class="btn btn-primary"
              id="contactFormSubmit"
              type="submit"
            >
              Send message
            </button>
          </div>
          <div
            class="formElementsStyles__ButtonWrapper-sc-5xvoh1-1 karwgk"
          >
            <button
              aria-label="Clear"
              class="btn btn-primary hollow"
              id="contactFormSubmit"
              type="reset"
            >
              Clear
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
